distro/run/core/src/main/java/org/camunda/bpm/run/CamundaBpmRunProcessEngineConfiguration.java

```diff
@@ -16,38 +16,40 @@
  */
 package org.camunda.bpm.run;
 
-import jakarta.inject.Inject;
-import java.io.File;
-import java.io.IOException;
-import java.util.Set;
-
 import org.camunda.bpm.engine.ProcessEngineException;
+import org.camunda.bpm.engine.impl.cfg.CompositeProcessEnginePlugin;
+import org.camunda.bpm.engine.impl.cfg.ProcessEnginePlugin;
 import org.camunda.bpm.engine.impl.diagnostics.CamundaIntegration;
 import org.camunda.bpm.engine.spring.SpringProcessEngineConfiguration;
-import org.springframework.core.env.Environment;
+import org.camunda.bpm.run.property.CamundaBpmRunProcessEnginePluginProperty;
+import org.camunda.bpm.run.property.CamundaBpmRunProperties;
+import org.camunda.bpm.run.utils.CamundaBpmRunProcessEnginePluginHelper;
 import org.springframework.core.io.Resource;
 
-public class CamundaBpmRunProcessEngineConfiguration extends SpringProcessEngineConfiguration {
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
 
+public class CamundaBpmRunProcessEngineConfiguration extends SpringProcessEngineConfiguration {
 
-  @Inject
-  private Environment env;
+  private final String normalisedDeploymentDir;
 
-  public CamundaBpmRunProcessEngineConfiguration() {
-    setDeployChangedOnly(true);
+  public CamundaBpmRunProcessEngineConfiguration(CamundaBpmRunProperties runProperties,
+                                                 String normalisedDeploymentDir,
+                                                 boolean deployChangedOnly,
+                                                 List<ProcessEnginePlugin> processEnginePlugins) {
+    this.normalisedDeploymentDir = normalisedDeploymentDir;
+    setDeployChangedOnly(deployChangedOnly);
+    configureProcessEnginePlugins(processEnginePlugins, runProperties);
   }
 
   @Override
   protected String getFileResourceName(Resource resource) {
     // only path relative to the root deployment directory as identifier to
     // prevent re-deployments when the path changes (e.g. distro is moved)
     try {
-      String deploymentDir = env.getProperty(CamundaBpmRunDeploymentConfiguration.CAMUNDA_DEPLOYMENT_DIR_PROPERTY);
-      if(File.separator.equals("\\")) {
```

Comment:
**Reason**: The normalising of the directory  behaviour is extracted into the `CamundaBpmRunDeploymentConfiguration` where it belongs (see [getNormalisedDeploymentsDir](https://github.com/camunda/camunda-bpm-platform/blob/23cc398ae30e011a2fffd081a29c542c8807bbc0/distro/run/core/src/main/java/org/camunda/bpm/run/CamundaBpmRunDeploymentConfiguration.java#L56-L63))